### PR TITLE
Add python app support

### PR DESCRIPTION
**Pull request by jmetzmeier**
This is still a work in progress. The following still needs to be done before merging:

* Test with more frameworks (currently, only Flask has been tested)
* Validate python_version at compile time so a friendly error message can be presented instead of failing unexpectedly during "dev" or "build"
* Add documentation 

Feedback is welcome.

### CODE REVIEW

**Comment by jmetzmeier** [comment]
I don't think my second point about validating python_version will be required to merge. I check and ruby_version suffers from the same problem. I opened issue #162 to add support for compile-time validation of customization params.

File: builtin/app/python/data/aws-simple/build/build-python.sh.tpl [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+start on (filesystem)
+stop on runlevel [016]
+
+respawn
+setuid otto-app
+setgid otto-app
+chdir /srv/otto-app/src
+
+exec /srv/otto-app/virtualenv/bin/gunicorn -w 4 $PYTHON_ENTRYPOINT
```
Comment (jezdez) [inline builtin/app/python/data/aws-simple/build/build-python.sh.tpl]: number of workers would be a neat thing to make a config flag

**Comment by BuhtigithuB** [comment]
Can this be merged?

**Comment by jmetzmeier** [comment]
No, it can't be merged. When otto was updated to support plugins the structure of the built-in apps was changed. I need to go back and do the same refactoring that was done for the other apps to this one, I just haven't had the time. I will revisit it after the holidays.

**Pull request merged**